### DISCUSSION TITLE
What's the best way to install Jupyter for use with Julia, on a Mac?

### DISCUSSION THREAD

**Discussion by mpf** [quarto-dev/quarto-cli#8240] [Q&A]
I'm using Quarto mainly with Julia. There seems to be at least two ways to install Jupyter (and similarly for Jupyter extensions):
- outside of Julia via `python3 -m pip install jupyter` (which is what `quarto check` suggests)
- from within Julia via `Pkg.install('IJulia'); using IJulia; notebook()`

What's the suggested approach?

**Comment by mcanouil** [comment]
I would say the latter if you don't really use Python alone.
But if you are looking for "official" guidelines, there are none on this matter.

Note that in the end, the best approach for you is the approach that works for you and allows you to do what you want to do.

**Comment by mpf** [comment]
Thanks,  @mcanouil . I've chosen to install Jupyter from outside of Julia. This approach seems to work reliably for a new project:
```
virtualenv venv
source venv/bin/activate
julia --project=@. -e 'using Pkg; Pkg.add("IJulia")'
python3 -m pip install jupyter
python3 -m pip install jupyter-cache
```
